### ISSUE TITLE
NullPointer exception on running sketch on processing

### ISSUE THREAD

**Issue by cjdg** [sparkfun/SparkFun_GridEYE_Arduino_Library#1]
Im havin a lot of errors on Processing 3, some are nullpointer exceptions and map(NaN, 20, 40, 240, 360) called, which returns NaN (not a number)

**Comment by juichi** [comment]
I found that adding a small delay after the check to see if serial data was available stopped the NullPointer exception.  I do not have any idea why it fixed it and I think it is a kludge. My thinking is that there is something messy with the Processing 3 Serial handler, my computer or something.  I also ran this on an Wemos mini Lite so maybe that is a factor.   Oh I also changed my null terminator in the arduino code to '\0' to try to fix this, that is why I have a readStringUntil(00).  The initial code has 13 which is carriage return.

See below:
` if(myPort.available() > 64){`
`    delay(40);`  <---- add this.
`  myString = myPort.readStringUntil(00) ;`

**Comment by jmorsap** [comment]
I have experienced the same problem with Processing 3. I added a little check to make sure that the string is not empty before doing anything else in the draw() function.

```
if (myString !=null)
{
  String splitString[] = splitTokens(myString, ",");
  
......
```

Seems to solve it, but I have no idea why the string would be empty after the   `myString = myPort.readStringUntil(13);` command. I haven't used processing very much. Would the next line execute before the readStringUntil function has completed?

**Comment by jmorsap** [comment]
FYI, looks like the issue is with the line 
`if(myPort.available() > 64){`

There's actually something closer to 300 bytes available before the carriage return character is sent, but I think this varies as the data is sent as a string of floats, which I think can have different numbers of characters in the decimal value, so the length of bytes in the float string might change depending on what the data looks like. 

A better way for the code to be written might be to use something that reads each character as it comes in, adds it to a buffer, and checks if the character is the string terminator. 

```
      if (myPort.available()>0)
      {
      mybuff[i] = myPort.read();
i++; //I needs to be global
}
if(mybuff[i] == 13)
{
i = 0;
//parse the data out
//update the pixels
}
```

**Comment by PaulZC** [comment]
Hi @cjdg @juichi @jmorsap ,

Thank you for raising this issue - and apologies for the *VERY* slow reply!

I've merged @Suisse00 's PR #4 into v1.0.2. v1.0.2 will be released shortly. I know it has been _years_ since this issue was opened, but please do give v1.0.2 a try and let me know if you see any other issues.

Very best wishes,
Paul
